### PR TITLE
Added test for deleting attributes from a Variant and fixed bugs with outputs no wired

### PR DESCRIPTION
**Pull request by cglzaguilar**

### CODE REVIEW

File: source/core/Variants.cpp [hunk 1]
```diff
@@ -196,35 +204,59 @@ VIREO_FUNCTION_SIGNATURET(GetVariantAttributeAll, GetVariantAttributesAllParamBl
...
                     String* const* attributeNameStr = &(attributePair.first);
                     TypeRef attributeValue = attributePair.second;
-                    namesElementType->CopyData(attributeNameStr, pNamesInsert);
-                    if (attributeValue->Name().Compare(&TypeCommon::TypeVariant)) {
-                        attributeValue->CopyData(attributeValue->Begin(kPARead), pValuesInsert);
-                    } else {
-                        TypeRef variant = DefaultValueType::New(tm, attributeValue, true);
-                        variant->CopyData(attributeValue->Begin(kPARead), variant->Begin(kPAWrite));
-                        *reinterpret_cast<TypeRef *>(pValuesInsert) = variant;
+                    if (names)
```
[the remark this answers is not in the document]
Comment (cglzaguilar) [reply source/core/Variants.cpp]: Yes, I already have tests, that were submitted in the previous PR for this primitive.

File: source/core/Variants.cpp [hunk 2]
```diff
@@ -183,8 +183,16 @@ struct GetVariantAttributesAllParamBlock : public InstructionCore
 VIREO_FUNCTION_SIGNATURET(GetVariantAttributeAll, GetVariantAttributesAllParamBlock)
 {
     ErrorCluster *errPtr = _ParamPointer(ErrorClust);
-    TypedArrayCoreRef names = _Param(Names);
-    TypedArrayCoreRef values = _Param(Values);
+    TypedArrayCoreRef names = nullptr;
+    TypedArrayCoreRef values = nullptr;
+    if (_ParamPointer(Names))
+    {
```
[the remark this answers is not in the document]
Comment (cglzaguilar) [reply source/core/Variants.cpp]: Done.

File: source/core/Variants.cpp [hunk 3]
```diff
@@ -183,8 +183,16 @@ struct GetVariantAttributesAllParamBlock : public InstructionCore
...
+        names = _Param(Names);
+    }
+    if (_ParamPointer(Values))
```
[the remark this answers is not in the document]
Comment (cglzaguilar) [reply source/core/Variants.cpp]: Done.

File: source/core/Variants.cpp [hunk 4]
```diff
@@ -183,8 +183,16 @@ struct GetVariantAttributesAllParamBlock : public InstructionCore
...
+    {
+        values = _Param(Values);
+    }
     bool bResetOutputArrays = true;
     if (!errPtr || !errPtr->status) {
```
[the remark this answers is not in the document]
Comment (cglzaguilar) [reply source/core/Variants.cpp]: Done.

File: source/core/Variants.cpp [hunk 5]
```diff
@@ -252,25 +285,31 @@ VIREO_FUNCTION_SIGNATURET(DeleteVariantAttribute, DeleteVariantAttributeParamBlo
...
                     const auto attributeMapIterator = attributeMap->find(*name);
                     if (attributeMapIterator != attributeMap->end()) {
-                        _Param(Found) = true;
+                        found = true;
                         attributeMapIterator->first->Delete(attributeMapIterator->first);
                         attributeMap->erase(attributeMapIterator);
                     }
                 }
             }
+        } else if (clearAllAttributes) {  // For LabVIEW NXG Compatibility
```
[the remark this answers is not in the document]
Comment (cglzaguilar) [reply source/core/Variants.cpp]: Removed comment. And yes that is the behavior in both CG and NXG.

File: source/core/Variants.cpp [hunk 6]
```diff
@@ -252,25 +285,31 @@ VIREO_FUNCTION_SIGNATURET(DeleteVariantAttribute, DeleteVariantAttributeParamBlo
...
+            found = true;
         }
     }
+    if (_ParamPointer(Found))
+    {
```
[the remark this answers is not in the document]
Comment (cglzaguilar) [reply source/core/Variants.cpp]: Done

File: source/include/Variants.h [hunk 7]
```diff
@@ -59,6 +59,9 @@ class VariantAttributeManager {
         CleanUp();
     }
 };
+
+enum { kVariantArgErr = 1, kVariantIncompatibleType = 91};
```
Comment (sanmut) [inline source/include/Variants.h]: I think this is fine for now. But, please create a github issue to consolidate all LV errors to a header file.

File: source/core/Variants.cpp [hunk 8]
```diff
@@ -167,16 +166,15 @@ VIREO_FUNCTION_SIGNATURET(GetVariantAttribute, GetVariantAttributeParamBlock)
                     found = true;
                     value->_paramType->CopyData(foundValue->Begin(kPARead), value->_pData);
                 } else {
-                    if (errPtr) {
-                        errPtr->SetErrorAndAppendCallChain(true, 91, "Get Variant Attribute");  // Incorrect type for default value for the attribute
+                    if (errPtr) {  // Incorrect type for default attribute value
```
Comment (sanmut) [inline source/core/Variants.cpp]: This is fine here again. Please add to the same github issue to define these as string constants and consolidate. Duplicating it just asks for misspelling and causing error call chain to fail in tests.

File: source/core/Variants.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (cglzaguilar) [reply source/core/Variants.cpp]: While doing testing I didn't find any scenario that generated an error.

File: source/core/Variants.cpp [hunk 9]
```diff
@@ -196,35 +204,59 @@ VIREO_FUNCTION_SIGNATURET(GetVariantAttributeAll, GetVariantAttributesAllParamBl
...
-                const Int32 namesAQSize = namesElementType->TopAQSize();
-                const Int32 valuesAQSize = valuesElementType->TopAQSize();
+                AQBlock1* pNamesInsert = nullptr;
+                AQBlock1* pValuesInsert = nullptr;
+                TypeRef namesElementType = nullptr;
+                TypeRef valuesElementType = nullptr;
+                Int32 namesAQSize = 0;
+                Int32 valuesAQSize = 0;
+                if (names)
+                {
```
[the remark this answers is not in the document]
Comment (cglzaguilar) [reply source/core/Variants.cpp]: Done

File: source/include/Variants.h (hunk 7 above)
Comment (cglzaguilar) [reply source/include/Variants.h]: Good point. There is already an issue for this: https://github.com/ni/VireoSDK/issues/384

File: source/core/Variants.cpp (hunk 8 above)
Comment (cglzaguilar) [reply source/core/Variants.cpp]: Good point. Updated the following issue: https://github.com/ni/VireoSDK/issues/384

File: source/include/Variants.h (hunk 7 above)
Comment (cglzaguilar) [reply source/include/Variants.h]: There was already an issue for this: https://github.com/ni/VireoSDK/issues/384

**Pull request merged**